src/Microsoft.OData.Core/Json/JsonWriter.cs

```diff
@@ -392,6 +398,7 @@ public void WriteRawValue(string rawValue)
         /// </summary>
         public void Flush()
         {
+            BufferUtils.ReturnToBuffer(this.ArrayPool, this.buffer);
             this.writer.Flush();

```

Comment:
Should we return to pool on Flush or Dispose?

Reply:
@mikepizzo I changed to use the "Dispose".